spec.html

```diff
@@ -30460,21 +30438,7 @@ <h1>The %StringIteratorPrototype% Object</h1>
         <emu-clause id="sec-%stringiteratorprototype%.next">
           <h1>%StringIteratorPrototype%.next ( )</h1>
           <emu-alg>
-            1. Let _O_ be the *this* value.
-            1. If Type(_O_) is not Object, throw a *TypeError* exception.
-            1. If _O_ does not have all of the internal slots of a String Iterator Instance (<emu-xref href="#sec-properties-of-string-iterator-instances"></emu-xref>), throw a *TypeError* exception.
-            1. Let _S_ be _O_.[[IteratedString]].
-            1. If _S_ is *undefined*, return CreateIterResultObject(*undefined*, *true*).
-            1. Let _position_ be _O_.[[StringNextIndex]].
-            1. Let _len_ be the length of _S_.
-            1. If _position_ &ge; _len_, then
-              1. Set _O_.[[IteratedString]] to *undefined*.
-              1. Return CreateIterResultObject(*undefined*, *true*).
-            1. Let _cp_ be ! CodePointAt(_S_, _position_).
-            1. Let _nextIndex_ be _position_ + _cp_.[[CodeUnitCount]].
-            1. Let _resultString_ be the substring of _S_ from _position_ to _nextIndex_.
-            1. Set _O_.[[StringNextIndex]] to _nextIndex_.
-            1. Return CreateIterResultObject(_resultString_, *false*).
+            1. Return ? GeneratorResume(*this* value, *undefined*, ~String Iterator~).
```

Comment:
Hmm, this represents what `yield` expression gets. So I think it must be a runtime value instead of `~empty~`

Reply:
There is no `yield` expression involved in this; it's purely a spec-internal algorithm. In particular, this call to GeneratorResume corresponds to the `1. Perform ? Yield(_resultString_).` step in the abstract closure created in `String.prototype [ @@iterator ] ( )`, which does not use the value inside of the completion record created by this call to `GeneratorResume` - it just checks that the completion is not abrupt.

Reply:
But `GeneratorResume` is also used in userland generator functions. I don't feel it is necessary to make difference here

Reply:
I don't think _all_ calls to `GeneratorResume` should pass `~empty~`, just these specific, strictly internal ones. I think it's important precisely because it emphasizes that these are not going to end up resuming an actual `yield` expression, which is otherwise not obvious here.

Reply:
Yeah, I know, normal `GeneratorResume` will pass what it received for `next(value)` call. If we pass a spec value here, should `GeneratorResume` be changed respectively to accept spec value?

Reply:
There's no changes to `GeneratorResume` necessary.